rotkehlchen/chain/ethereum/manager.py

```diff
@@ -606,6 +611,79 @@ def _get_code(self, web3: Optional[Web3], account: ChecksumEthAddress) -> str:
 
         return hex_or_bytes_to_str(web3.eth.getCode(account))
 
+    def _ens_reverse_lookup(
+            self,
+            web3: Optional[Web3],
+            reversed_addr: ChecksumEthAddress,
+            blockchain: SupportedBlockchain = SupportedBlockchain.ETHEREUM,
+    ) -> Optional[str]:
+        """Performs an ENS reverse lookup on the given address
+
+    May raise:
+        - RemoteError if Etherscan is used and there is a problem querying it or
+        - InputError if the address is not a valid wallet address
+        """
+        try:
+            reversed_domain = address_to_reverse_domain(reversed_addr)
+        except (TypeError, ValueError) as e:
+            raise InputError(str(e)) from e
+
+        normalized_domain_name = normalize_name(reversed_domain)
+        arguments = [normal_name_to_hash(normalized_domain_name)]
+        resolver_addr = self._call_contract(
```

Comment:
Why do you use `_call_contract` instead of `call_contract`? If you look at the code you will see that `call_contract` uses `query` that has extra logic for error handling that you don't do here. Is true that you have fewer functions calls but in the other hand you can have unhandled errors that would be properly cached but you could miss the opportunity to use other nodes

Reply:
As I thought, it is not needed there because _ens_lookup method also uses _call_contract

Reply:
No no, `ens lookup` is another method. I've been doing `ens reverse lookup` and took `ens lookup` as a reference. So maybe I don't understand something but why should we use a different structure in very similar methods?

Reply:
Maybe I misunderstand something but in my opinion this function does exactly the same, doesn't it?

Reply:
yes you are doing good with the one look I had. Just make sure to astract common code as I said elsewhere.